collector/asp_net_application.go

```diff
@@ -0,0 +1,160 @@
+package collector
+
+import (
+	"github.com/prometheus-community/windows_exporter/log"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/yusufpapurcu/wmi"
+)
+
+func init() {
+	registerCollector("asp_net_application", newasp_net_applicationCollector) // TODO: Add any perflib dependencies here
+}
+
+// A asp_net_applicationCollector is a Prometheus collector for WMI Win32_PerfRawData_ASPNET_ASPNETApplications metrics
+type asp_net_applicationCollector struct {
+	ErrorsDuringExecution                *prometheus.Desc
+	ErrorsTotalPerSec                    *prometheus.Desc
+	ManagedMemoryUsedestimated           *prometheus.Desc
+	PercentManagedProcessorTimeestimated *prometheus.Desc
+	RequestsExecuting                    *prometheus.Desc
+	RequestsInApplicationQueue           *prometheus.Desc
+	RequestsPerSec                       *prometheus.Desc
+	RequestWaitTime                      *prometheus.Desc
+}
+
+func newasp_net_applicationCollector() (Collector, error) {
+	const subsystem = "asp_net_application"
+	return &asp_net_applicationCollector{
+		ErrorsDuringExecution: prometheus.NewDesc(
+			prometheus.BuildFQName(Namespace, subsystem, "errors_during_execution"),
+			"The total number of errors that occur during the execution of an HTTP request, excluding parser and compilation errors.",
+			nil,
+			nil,
+		),
+		ErrorsTotalPerSec: prometheus.NewDesc(
+			prometheus.BuildFQName(Namespace, subsystem, "errors_total_per_sec"),
```

Comment:
If this is a counter it'll need a `_total` suffix. Description will need to updated to reflect the metric nature, being the total number of errors.